plugins/modules/ndo_interface_setting.py

```diff
@@ -0,0 +1,808 @@
+            cdp_admin_state=dict(type="str", choices=["enabled", "disabled"]),
+            lldp=dict(type="str", choices=["enabled", "disabled"]),
+            domains=dict(type="list", elements="str"),
+            port_channel_mode=dict(type="str", choices=list(PORT_CHANNEL_MODE_MAP.keys())),
+            min_links=dict(type="int"),
+            max_links=dict(type="int"),
+            control=dict(type="list", elements="str", choices=list(CONTROL_MAP.keys())),
+            load_balance_hashing=dict(type="str", choices=list(LOAD_BALANCE_HASHING_MAP.keys())),
+            sync_e=dict(type="str"),
+            link_level_debounce_interval=dict(type="int"),
+            link_level_bring_up_delay=dict(type="int"),
+            link_level_fec=dict(type="str", choices=list(LINK_LEVEL_FEC_MAP.keys())),
+            l2_interface_qin_q=dict(type="str", choices=list(L2_INTERFACE_QIN_Q_MAP.keys())),
+            l2_interface_reflective_relay=dict(type="str", choices=["enabled", "disabled"]),
+            lldp_transmit_state=dict(type="str", choices=["enabled", "disabled"]),
+            lldp_receive_state=dict(type="str", choices=["enabled", "disabled"]),
+            stp_bpdu_filter=dict(type="str", choices=["enabled", "disabled"]),
+            stp_bpdu_guard=dict(type="str", choices=["enabled", "disabled"]),
+            llfc_transmit_state=dict(type="str", choices=["enabled", "disabled"]),
+            llfc_receive_state=dict(type="str", choices=["enabled", "disabled"]),
+            mcp_admin_state=dict(type="str", choices=["enabled", "disabled"]),
+            mcp_strict_mode=dict(type="str", choices=["on", "off"]),
+            mcp_initial_delay_time=dict(type="int"),
+            mcp_transmission_frequency_sec=dict(type="int"),
+            mcp_transmission_frequency_msec=dict(type="int"),
+            mcp_grace_period_sec=dict(type="int"),
+            mcp_grace_period_msec=dict(type="int"),
+            pfc_admin_state=dict(type="str", choices=["on", "off", "auto"]),
+            access_mac_sec_policy=dict(type="str"),
+            state=dict(type="str", choices=["absent", "query", "present"], default="query"),
+        )
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "present", ["interface_policy_group", "interface_type"]],
+            ["state", "absent", ["interface_policy_group", "interface_policy_group_uuid"], True],
+        ],
+    )
+
+    mso = MSOModule(module)
+
+    template = module.params.get("template")
+    interface_policy_group = module.params.get("interface_policy_group")
+    interface_policy_group_uuid = module.params.get("interface_policy_group_uuid")
+    description = module.params.get("description")
+    interface_type = module.params.get("interface_type")  # .replace("_", "")
+    if interface_type == "pc_vpc":
+        interface_type = "portchannel"
```

Comment:
should we use a map like we do for most of these settings? module.params.get("interface_type") can then be passed into the map. See example: https://github.com/CiscoDevNet/ansible-mso/pull/519/files#diff-94c57da0a958768c4db919f7f7a3ade8910194ba4dfc23eb83397cee0e6a56eeR494

Reply:
same applies for the other conditionals, where I see for some you actually do this already, like control

Reply:
As discussed, replace will be used for few arguments instead of creating map, since they have only 2 arguments and only one of them needs translation.
- the arguments are : interface_type, auto_negotiation, vlan_scope